src/app/modules/user/user.controller.ts

```diff
@@ -35,31 +38,73 @@ export const handleContactSendOtp: RequestHandler<
   // Guard against user updating for a different user, or if user is not logged
   // in.
   if (!sessionUserId || sessionUserId !== userId) {
-    return res.status(StatusCodes.UNAUTHORIZED).send('User is unauthorized.')
+    return res
+      .status(StatusCodes.UNAUTHORIZED)
+      .json({ message: 'User is unauthorized.' })
   }
 
-  try {
-    const generatedOtp = await createContactOtp(userId, contact)
-    await SmsFactory.sendAdminContactOtp(contact, generatedOtp, userId)
+  const logMeta = {
+    action: 'handleContactSendOtp',
+    userId,
+    ip: getRequestIp(req),
+  }
 
-    return res.sendStatus(StatusCodes.OK)
-  } catch (err) {
-    // TODO(#193): Send different error messages according to error.
-    return res.status(StatusCodes.BAD_REQUEST).send(err.message)
+  // Step 1: Create OTP for contact verification.
+  const createResult = await createContactOtp(userId, contact)
+
+  // Error creating OTP.
+  if (createResult.isErr()) {
+    const { error } = createResult
+    logger.error({
+      message: 'Error creating contact verification OTP',
+      meta: logMeta,
+      error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(error)
+    return res.status(statusCode).json({ message: errorMessage })
```

Comment:
same here, not a backwards-compatible change right?

Reply:
the previous plan was to make all these backwards-incompatible changes at the same time as we migrate our APIs to a more RESTful hierarchy. but I'm not opposed to doing this now, as long as we're okay with a few clients potentially seeing an ugly JSON error message.

Reply:
Nice catch, thanks! Converting to `.json(string)` implementation without the `{message:string}` shape. This will have no backwards incompatibility

Reply:
damn, I was a clown and didn't realise you could `.json(string)` looool